app/uk/gov/hmrc/exports/services/mapping/CachingMappingHelper.scala

```diff
@@ -23,21 +23,26 @@ import uk.gov.hmrc.wco.dec._
 class CachingMappingHelper {
   val defaultMeasureCode = "KGM"
 
-  def commodityFromExportItem(exportItem: ExportItem): Commodity = {
-    def removeCarriageReturns(description: String): String = description.replaceAll("(\\r\\n)|\\r|\\n", " ")
+  def commodityFromExportItem(exportItem: ExportItem): Option[Commodity] = {
+    def removeCarriageReturns(description: Option[String]): Option[String] = description.map(_.replaceAll("(\\r\\n)|\\r|\\n", " "))
 
-    Commodity(
-      description = exportItem.commodityDetails.map(commodityDetails => removeCarriageReturns((commodityDetails.descriptionOfGoods))),
-      classifications = getClassifications(
-        exportItem.commodityDetails.flatMap(_.combinedNomenclatureCode),
-        exportItem.cusCode.flatMap(_.cusCode),
-        exportItem.nactCodes.map(_.map(_.nactCode)).getOrElse(List.empty),
-        exportItem.taricCodes.map(_.map(_.taricCode)).getOrElse(List.empty)
-      ),
-      dangerousGoods = exportItem.dangerousGoodsCode.flatMap(_.dangerousGoodsCode).map(code => Seq(DangerousGoods(Some(code)))).getOrElse(Seq.empty)
-    )
+    Some(
+      Commodity(
+        description = exportItem.commodityDetails.flatMap(commodityDetails => removeCarriageReturns(commodityDetails.descriptionOfGoods)),
+        classifications = getClassifications(
+          exportItem.commodityDetails.flatMap(_.combinedNomenclatureCode),
+          exportItem.cusCode.flatMap(_.cusCode),
+          exportItem.nactCodes.map(_.map(_.nactCode)).getOrElse(List.empty),
+          exportItem.taricCodes.map(_.map(_.taricCode)).getOrElse(List.empty)
+        ),
+        dangerousGoods = exportItem.dangerousGoodsCode.flatMap(_.dangerousGoodsCode).map(code => Seq(DangerousGoods(Some(code)))).getOrElse(Seq.empty)
+      )
+    ).filter(isCommodityEmpty)
   }
 
+  private def isCommodityEmpty(commodity: Commodity): Boolean =
```

Comment:
One final suggestion.   Should this method name not be negated?  I.e. `isCommodityNonEmpty'?

And good point about not adding method to wco model.  Why will we have to remove the wco models BTW?

Reply:
Yes it should be name `isCommodityNonEmpty` :+1: 
The Models were created in `wco-dec` library and we should really not used them, as agreed a few months back, but somehow these the item level models have not been migrated.